tools/aiecc/aiecc/main.py

```diff
@@ -221,11 +221,15 @@ async def process_arm_cgen(self):
       # Generate the included host interface
       file_physical = os.path.join(self.tmpdirname, 'input_physical.mlir')
       await self.do_call(task, ['aie-opt', '--aie-create-pathfinder-flows', '--aie-lower-broadcast-packet', '--aie-create-packet-flows', '--aie-lower-multicast', self.file_with_addresses, '-o', file_physical]);
-      file_inc_cpp = os.path.join(self.tmpdirname, 'aie_inc.cpp')
-      if(opts.xaie == 1):
-          await self.do_call(task, ['aie-translate', '--aie-generate-xaie', '--xaie-target=v1', file_physical, '-o', file_inc_cpp])
+      if(opts.airbin):
+        file_airbin = os.path.join(self.tmpdirname, 'air.bin')
+        await self.do_call(task, ['aie-translate', '--aie-generate-airbin', file_physical, '-o', file_airbin])
       else:
-          await self.do_call(task, ['aie-translate', '--aie-generate-xaie', '--xaie-target=v2', file_physical, '-o', file_inc_cpp])
+        file_inc_cpp = os.path.join(self.tmpdirname, 'aie_inc.cpp')
+        if(opts.xaie == 1):
```

Comment:
As I read this, you're no longer generating aie_inc.cpp... But there's a bunch of host APIs in there we probably can't just drop.

Reply:
For AIR we don't need those APIs when we are using airbin, air_inc.cpp is only used for configuration via libXAIE

Reply:
So all the tests run with airbin enabled?  There are more host APIs generated than just configuration.

Reply:
Which tests? This works for the mlir-air tests using airbin, it has not been tested on every air test/example yet. It is not intended as a back end for mlir-aie tests without the air runtime

Reply:
But seems like it very easily could be...